tests/unit/lib/SGMatrix_unittest.cc

```diff
@@ -761,3 +761,37 @@ TEST(SGMatrixTest,iterator)
 	for (auto v: mat)
 		EXPECT_EQ(mat[index++], v);
 }
+
+TEST(SGMatrixTest,column_iterator)
+{
+	constexpr index_t size=5;
+	SGMatrix<float64_t> mat(size, size);
+	linalg::range_fill(mat, 1.0);
+
+	SGMatrix<float64_t>::column_iterator col_itr(mat);
```

Comment:
instead of this, should I use 
``` 
auto begin_col = mat.begin_column();
auto end_col = mat.end_column();
EXPECT_EQ(mat.num_cols, std::distance(begin_col, end_col));
EXPECT_TRUE(begin_col != end_col);
``` 

Sounds better to me (I compiled and tested it, it works )

Reply:
Okay, I implemented the same.